test/compilable/test18099.d

```diff
@@ -0,0 +1,19 @@
+
+/* REQUIRED_ARGS: -betterC
+ */
+
+// https://issues.dlang.org/show_bug.cgi?id=18099
+
+struct D
+{
+    static struct V
+    {
+	~this() { }
+    }
+
+    V get()
+    {
+	V v;
+	return v;
+    }
+}
```

Comment:
The formatting still looks weird (at least in the GitHub web interface).  Maybe there are some tab characters there that shouldn't be there.

Reply:
fixed the tabs